Library/String+Currency.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+import UIKit.UIColor
+import UIKit.UIFont
+
+public func attributedCurrencyString(
+  currencySymbol: String,
+  amount: Double,
+  fractionDigits: UInt,
+  font: UIFont,
+  superscriptFont: UIFont,
+  foregroundColor: UIColor
+  ) -> NSAttributedString {
+  // Drop decimal places
+  let formattedString = String(format: "\(currencySymbol)%.\(fractionDigits)f", amount)
+
+  let attributedString = NSMutableAttributedString(string: formattedString)
+  let franctionDigitsAndSeparator = Int(fractionDigits == 0 ? fractionDigits : fractionDigits + 1)
+
+  // Calculate prefix and suffix ranges
+  let prefix = NSRange(location: 0, length: currencySymbol.count)
+  let range = NSRange(location: 0, length: attributedString.length)
+  let suffix = NSRange(
+    location: attributedString.length - franctionDigitsAndSeparator,
+    length: franctionDigitsAndSeparator
+  )
+
+  // Calculate vertical offset based on the height of a capital character of the two fonts
+  let maxCapHeight: CGFloat = max(font.capHeight, superscriptFont.capHeight)
+  let minCapHeight: CGFloat = min(font.capHeight, superscriptFont.capHeight)
+  let multiplier: CGFloat = font.capHeight > superscriptFont.capHeight ? 1 : 0
+  let baselineOffset = NSNumber(value: Float(multiplier * (maxCapHeight - minCapHeight)))
+
+  // Set font for the whole string
```

Comment:
Any benefit to grouping things like this by range?

```swift
// Set font and foreground color for the whole string
attributedString.setAttributes([
  NSAttributedString.Key.font: font,
  NSAttributedString.Key.foregroundColor: foregroundColor
], range: range)

// Replace prefix font and offset vertically from the baseline
attributedString.setAttributes([
  NSAttributedString.Key.font: superscriptFont,
  NSAttributedString.Key.baselineOffset: baselineOffset
], range: prefix)

// Replace suffix font and offset vertically from the baseline
attributedString.setAttributes([
  NSAttributedString.Key.font: superscriptFont,
  NSAttributedString.Key.baselineOffset: baselineOffset
], range: suffix)
```

Reply:
Could do! Thought it was more readable to do 1 command per line but yeah, I'm not opposed to this.

Reply:
The only downside to that is that `setAttributes` replaces all the previously set attributes so we have to additionally also set the foreground color like so:

```
// Set font and foreground color for the whole string
attributedString.setAttributes([
  NSAttributedString.Key.font: font,
  NSAttributedString.Key.foregroundColor: foregroundColor
], range: range)

// Replace prefix font and offset vertically from the baseline
attributedString.setAttributes([
  NSAttributedString.Key.font: superscriptFont,
  NSAttributedString.Key.baselineOffset: baselineOffset,
  NSAttributedString.Key.foregroundColor: foregroundColor
], range: prefix)

// Replace suffix font and offset vertically from the baseline
attributedString.setAttributes([
  NSAttributedString.Key.font: superscriptFont,
  NSAttributedString.Key.baselineOffset: baselineOffset,
  NSAttributedString.Key.foregroundColor: foregroundColor
], range: suffix)
```

Reply:
Ah, no that's not worth it, I'd leave as is 👍